api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ViralLoadIntensiveMonitoringCohortQueries.java

```diff
@@ -0,0 +1,2685 @@
+            + "                              AND e.encounter_type = ${53} "
+            + "                              AND o.concept_id IN (${1982},${6332}) "
+            + "                              AND o.value_coded = ${1065} "
+            + "                              AND o.obs_datetime = final_min.final_min_date "
+            + "                            UNION "
+            + "                            SELECT p.patient_id "
+            + "                            FROM patient p "
+            + "                                     INNER JOIN encounter e ON e.patient_id = p.patient_id "
+            + "                                     INNER JOIN obs o ON e.encounter_id = o.encounter_id "
+            + "                            WHERE p.voided = 0 "
+            + "                              AND o.voided = 0 "
+            + "                              AND e.voided = 0 "
+            + "                              AND e.encounter_type = ${6} "
+            + "                              AND o.concept_id IN (${1982},${6332}) "
+            + "                              AND o.value_coded = ${1065} "
+            + "                              AND e.encounter_datetime = final_min.final_min_date)";
+
+    StringSubstitutor sb = new StringSubstitutor(map);
+    sqlCohortDefinition.setQuery(sb.replace(sql));
+    return sqlCohortDefinition;
+  }
+
+  public CohortDefinition getFirstLineArt() {
+    SqlCohortDefinition sqlCohortDefinition = new SqlCohortDefinition();
+    sqlCohortDefinition.setName("Linha Terapeutica");
+    sqlCohortDefinition.addParameter(new Parameter("startDate", "startDate", Date.class));
+    sqlCohortDefinition.addParameter(new Parameter("endDate", "endDate", Date.class));
+    sqlCohortDefinition.addParameter(new Parameter("location", "location", Location.class));
+
+    Map<String, Integer> map = new HashMap<>();
+    map.put("6", hivMetadata.getAdultoSeguimentoEncounterType().getEncounterTypeId());
+    map.put("21151", hivMetadata.getTherapeuticLineConcept().getConceptId());
+    map.put("21150", hivMetadata.getFirstLineConcept().getConceptId());
+
+    String sql =
+        " SELECT p.patient_id  "
+            + " FROM patient p INNER  JOIN ("
+            + " SELECT p.patient_id , MAX(encounter_datetime) AS last_date "
+            + " FROM patient p "
+            + "    INNER JOIN encounter e ON e.patient_id = p.patient_id "
+            + "    INNER JOIN obs o ON e.encounter_id = o.encounter_id "
+            + " WHERE p.voided = 0 "
+            + "  AND o.voided = 0 "
+            + "  AND e.voided = 0 "
+            + "  AND e.encounter_type = ${6} "
+            + "  AND o.concept_id = ${21151} "
+            + "  AND o.value_coded = ${21150} "
+            + "  AND e.encounter_datetime BETWEEN :startDate AND :endDate "
+            + "  AND e.location_id = :location )  ultima ON ultima.patient_id = p.patient_id";
```

Comment:
it was   mistake  i  implemented with max()  instead of  min()   as the spec    says : Filter all patients who are in the `first` line of ART in the evaluation period as

Reply:
corrected